sssd_test_framework/utils/sbus.py

```diff
@@ -0,0 +1,476 @@
+"""Manage D-Bus operations"""
+
+from __future__ import annotations
+
+import xml.etree.ElementTree as xml
+from abc import ABC
+from typing import Any, Final
+
+from pytest_mh import MultihostHost, MultihostUtility
+from pytest_mh.ssh import SSHProcessResult
+
+from sssd_test_framework.utils.dbus_types import (
+    DBUSResult,
+    DBUSSignatureReader,
+    DBUSType,
+    DBUSTypeString,
+    DBUSTypeVariant,
+)
+
+__all__ = [
+    "SBUSUtils",
+    "DBUSKnownBus",
+]
+
+
+class DBUSKnownBus(str):
+    """
+    Weel-known D-Bus buses.
+    """
+
+    SYSTEM: Final = "--SYSTEM--"
+    """The system bus"""
+
+    SESSION: Final = "--SESSION--"
+    """The session bus"""
+
+    MONITOR: Final = "unix:path=/var/lib/sss/pipes/private/sbus-monitor"
+    """Monitor's private bus"""
+
+    @classmethod
+    def forDomain(cls, name: str):
+        """Produce the bus for the given domain"""
+        return cls(f"unix:path=/var/lib/sss/pipes/private/sbus-dp_{name}")
+
+
+class ProxyObject(ABC):
+    """
+    Abstract class representing a D-Bus object (method or property).
+
+    Corresponding concrete classes are provided for their instantiation.
+    """
+
+    def __init__(
+        self,
+        host: MultihostHost,
+        dest: str,
+        objpath: str,
+        bus: str,
+        /,
+        interface: str | None = None,
+        child: xml.Element | None = None,
+    ):
+        """
+        Constructor to be called by the subclasses either explicitely or by inheritance.
+        """
+        match bus:
+            case DBUSKnownBus.SYSTEM:
+                _bus_param = "--system"
+            case DBUSKnownBus.SESSION:
+                _bus_param = "--session"
+            case _:
+                _bus_param = "--bus=" + bus
+
+        self.host = host
+        self.dest = dest
+        self.objpath = objpath
+        self.bus = bus
+        self.interface = objpath[1:].replace("/", ".") if interface is None else interface
+        self.child = child
+        self.cmd = f"dbus-send {_bus_param} --print-reply --dest={dest} {self.objpath}"
+
+    def _run(self, command: str, *args):
+        """
+        Run dbus-send over ssh on the remote host.
+        """
+        cmd = self.cmd + f" {command}"
+        for arg in args:
+            cmd += " " + arg.param()
+
+        rslt = self.host.ssh.run(cmd)
```

Comment:
Please, use `result` instead of `rslt`.

This call will automatically raise an error with all the information you can get if rc != 0. So you don't have to handler that in callers of this method. You can just return modified stdout instead of modifying the result.

Reply:
Done